### PR TITLE
feat(storage): task progress track num_progress_key

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Current pr has further improved the changes at https://github.com/risingwavelabs/risingwave/pull/9719

1. task progress track num_progress_key 
2. check wheather task duration is too long
3. more log for compaction progress

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by Gun9niR — APPROVED** [review]
LGTM

**Review by zwang28 — APPROVED** [review]

File: src/storage/src/hummock/compactor/mod.rs [hunk 1]
```diff
@@ -655,7 +662,15 @@ impl Compactor {
         let mut last_table_stats = TableStats::default();
         let mut last_table_id = None;
         let mut compaction_statistics = CompactionStatistics::default();
+        let mut progress_key_num: u64 = 0;
+        const PROGRESS_KEY_INTERVAL: u64 = 100;
         while iter.is_valid() {
+            progress_key_num += 1;
+
+            if let Some(task_progress) = task_progress.as_ref() && progress_key_num % PROGRESS_KEY_INTERVAL == 0 {
```
Comment (Little-Wallace) [inline src/storage/src/hummock/compactor/mod.rs:670]: suggest:
```rust
if let Some(task_progress) = task_progress.as_ref() && progress_key_num >= PROGRESS_KEY_INTERVAL {
   task_progress.inc_progress_key(progress_key_num);
   progress_key_num = 0
}
```

and then we can call this in the end of this function `task_progress.inc_progress_key(progress_key_num);`

**Review by Little-Wallace — APPROVED** [review]
rest LGTM

File: src/storage/src/hummock/compactor/mod.rs (hunk 1 above)
Comment (Li0k) [reply src/storage/src/hummock/compactor/mod.rs]: Good catch ! fix it

**Pull request merged**